splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceDateFunctionsTest.java

```diff
@@ -265,9 +268,41 @@ public void toChar() throws ParseException {
         String format = "MM/dd/yyyy";
         String compare = "06/24/2014";
         assertEquals(compare, SpliceDateFunctions.TO_CHAR(date, format));
+
         assertEquals(compare, SpliceDateFunctions.TIMESTAMP_TO_CHAR(timestamp, format));
     }
 
+    @Test
+    public void toCharInvalidFormat() throws ParseException, StandardException {
+        DateFormat formatter = new SimpleDateFormat("yyyy/MM/dd");
+        Date date = new Date(formatter.parse("2014/06/24").getTime());
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        Timestamp timestamp = new Timestamp(calendar.getTimeInMillis());
+        String format = "invalid_format";
+        String compare = "06.24.2014";
+
+        try {
+            SpliceDateFunctions.TO_CHAR(date, format);
+            Assert.fail("expected exception");
+        } catch(Exception e) {
+            Assert.assertEquals("ERROR 22018: Invalid character string format for type datetime.\n" +
+                    "Splice Machine Release: null\n" +
+                    "Splice Machine Version Hash: null\n" +
+                    "Splice Machine Build Time: null", e.toString());
```

Comment:
I would suggest checking only for the error message and not all the release/version/build to make this test more robust:

With:
```
import static org.hamcrest.Matchers.*;
```

```suggestion
                        Assert.assertThat(e.toString(), containsString("ERROR 22018: Invalid character string format for type datetime."));
```